### PR TITLE
remove odo catalog command

### LINKED ISSUE #5562: remove `odo catalog`
more info in #5521 

## Acceptance criteria 
- [x] test cases for the removed command should be removed, tests cases that use removed commands should be updated
- [x] commands that are going to be removed need to be removed together with their implementation and test cases
- [x] all help messages and error messages need to be reviewed and updated to make sure that it doesn't contain a reference to commands that no longer exist

### PR DESCRIPTION
**Pull request by kadel**
**What type of PR is this:**

/kind cleanup

**What does this PR do / why we need it:**
- remove `odo catalog` command 
- move devfile registry relevant code to pkg/registry
- removed page about `odo catalog` from docs
- updated docs for `odo init` to not mention `odo catalog`

There is still documentation that mentions `odo catalog` command but that in quick start page, setup or pages about other commands. Those should be addressed in PRs for that command, or in a separate effort to alight docs with v3-alpha1

**Which issue(s) this PR fixes:**

Fixes #5562

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: pkg/odo/genericclioptions/clientset/clientset.go [hunk 1]
```diff
@@ -123,24 +122,21 @@ func Fetch(command *cobra.Command) (*Clientset, error) {
...
-	}
 	if isDefined(command, DELETE_COMPONENT) {
 		dep.DeleteClient = _delete.NewDeleteComponentClient(dep.KubernetesClient)
 	}
 	if isDefined(command, DEPLOY) {
 		dep.DeployClient = deploy.NewDeployClient(dep.KubernetesClient)
 	}
 	if isDefined(command, INIT) {
-		dep.InitClient = _init.NewInitClient(dep.FS, dep.PreferenceClient, dep.RegistryClient, dep.CatalogClient)
+		dep.InitClient = _init.NewInitClient(dep.FS, dep.PreferenceClient, dep.RegistryClient)
```
Comment (feloy) [inline pkg/odo/genericclioptions/clientset/clientset.go:139]: You will need to remove the dependency of catalog in INIT entry in `subdeps`

**Review by feloy — CHANGES_REQUESTED** [review]

**Review by valaparthvi — CHANGES_REQUESTED** [review]
There are still remains in docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md. Other docs such as `create.md` and `service.md` still have references, but they can probably be removed in the relevant cleanup PRs.

**Comment by kadel** [comment]
> There are still remains in docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md. Other docs such as `create.md` and `service.md` still have references, but they can probably be removed in the relevant cleanup PRs.

Yes, those documents need to be completely reworked. It should be done in some separate PR that will focus only on docs. 
I just wanted to remove the most obvious parts of the documentation for this in this PR

**Comment by valaparthvi** [comment]
> > There are still remains in docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md. Other docs such as `create.md` and `service.md` still have references, but they can probably be removed in the relevant cleanup PRs.
> 
> Yes, those documents need to be completely reworked. It should be done in some separate PR that will focus only on docs. I just wanted to remove the most obvious parts of the documentation for this in this PR

You can still remove from the json-output.md because it is relevant and in the scope of your PR.

**Comment by kadel** [comment]
> > > There are still remains in docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md. Other docs such as `create.md` and `service.md` still have references, but they can probably be removed in the relevant cleanup PRs.
> > 
> > 
> > Yes, those documents need to be completely reworked. It should be done in some separate PR that will focus only on docs. I just wanted to remove the most obvious parts of the documentation for this in this PR
> 
> You can still remove from the json-output.md because it is relevant and in the scope of your PR.

Seem a little bit pointless as that whole document is not valid for v3. We need to remove that whole file, but that I think is not relevant to the original issue this PR tries to address. I'll do that in another PR

**Comment by valaparthvi** [comment]
> > > > There are still remains in docs/website/versioned_docs/version-3.0.0/command-reference/json-output.md. Other docs such as `create.md` and `service.md` still have references, but they can probably be removed in the relevant cleanup PRs.
> > > 
> > > 
> > > Yes, those documents need to be completely reworked. It should be done in some separate PR that will focus only on docs. I just wanted to remove the most obvious parts of the documentation for this in this PR
> > 
> > 
> > You can still remove from the json-output.md because it is relevant and in the scope of your PR.
> 
> Seem a little bit pointless as that whole document is not valid for v3. We need to remove that whole file, but that I think is not relevant to the original issue this PR tries to address. I'll do that in another PR
> 
> update: opened #5585

Oh, you're right. It completely skipped my mind that we haven't implemented JSON for v3. Sorry about this.

**Comment by dharmit** [comment]
```
Running oc with args [oc get pods --namespace cmd-devfile-push-test693jkd --selector=component=iwqqlu -o jsonpath={.items[*].metadata.name}]
[oc] iwqqlu-app-6f5f8747d9-74f7bRunning oc with args [oc exec iwqqlu-app-6f5f8747d9-74f7b -c runtime --namespace cmd-devfile-push-test693jkd -- ps -ef]
[oc] UID          PID    PPID  C STIME TTY          TIME CMD
[oc] 1003410+       1       0  0 18:06 ?        00:00:00 /opt/odo/bin/supervisord -c /opt/odo/conf/devfile-supervisor.conf
[oc] 1003410+      37       1  0 18:07 ?        00:00:00 /bin/sh /opt/odo/bin/devfile-command devrun
[oc] 1003410+      40      37  0 18:07 ?        00:00:00 sh -c cd /projects && mvn -Dmaven.repo.local=/data/cache/.m2/repository spring-boot:run
[oc] Signal 23 (URG) caught by ps (3.3.15).
[oc] ps:ps/display.c:66: please report this bug
[oc] command terminated with exit code 1
Deleting project: cmd-devfile-push-test693jkd 
```
Doesn't look like a real failure to me.

/test v4.10-integration-e2e
/approve

**Pull request merged**